azure-mgmt-sql/src/main/java/com/microsoft/azure/management/sql/implementation/SqlServerImpl.java

```diff
@@ -264,7 +268,7 @@ public SqlServerImpl withNewElasticPool(String elasticPoolName, ElasticPoolEditi
 
     @Override
     public SqlServerImpl withNewElasticPool(String elasticPoolName, ElasticPoolEditions elasticPoolEdition) {
-        return withNewElasticPool(elasticPoolName, elasticPoolEdition, null);
+        return withNewElasticPool(elasticPoolName, elasticPoolEdition, (String) null);
```

Comment:
background: not having a cast results in a compiler warning because of the ambiguous interpretation of the null. So I cast it explicitly to String, but it looks like the more proper cast to preserve the inferred logic is to cast to String[]. Added a commit to this PR addressing that.